### ISSUE TITLE
Issue about Variable `self.levels`.

### ISSUE THREAD

**Issue by ZhiyeTang** [city-super/Octree-GS#33]
In functions `self.create_from_pcd` and `self.load_ply_sparse_gaussian`, the method of determining `self.levels` makes a difference. Here is the code.

In `self.load_ply_sparse_gaussian`,
```
self.levels = torch.max(self._level) - torch.min(self._level) + 1
```
while in `self.create_from_pcd`, the adaptive level selection mechanism determines `self.levels` as
```
self.levels = torch.round(torch.log2(dist_max/dist_min)/math.log2(self.fork)).int().item() + 1
```

This seems to assign different values to `self.levels` for the same model when initializing and reloading a `gaussian_model`, which is unexpected.

**Comment by ZhiyeTang** [comment]
Furthermore, I noticed something strange. The released code is based on `Scaffold-GS`, "inheriting" the `warmup` training style they came up with. However, in the implemented class `Scene`, the loading logic from `ply_path` is modified, while in `Scaffold-GS`, there is:
![image](https://github.com/city-super/Octree-GS/assets/52972416/a7e11ef6-af66-4f6b-b1a6-5a70631bee5d)

**Comment by ZhiyeTang** [comment]
I also noticed something strange. `Scaffold-GS` and `Octree-GS` utilized a "warmup" training style. But in `Scaffold-GS`, setting `warmup` will replace the input point cloud with an already-trained point cloud, while in `Octree-GS` similar mechanism is missing in `scene/__init__.py: line 55-66`. However, setting `warmup` does promote the visual performance of `Octree-GS`.

**Comment by ZhiyeTang** [comment]
FYI, in `Scaffold-GS`, the code of replacement is as follows (`scene/__init_.py: line 54-70`)
```
        if not self.loaded_iter:
            if ply_path is not None:
                with open(ply_path, 'rb') as src_file, open(os.path.join(self.model_path, "input.ply") , 'wb') as dest_file:
                    dest_file.write(src_file.read())
            else:
                with open(scene_info.ply_path, 'rb') as src_file, open(os.path.join(self.model_path, "input.ply") , 'wb') as dest_file:
                    dest_file.write(src_file.read())
            json_cams = []
            camlist = []
            if scene_info.test_cameras:
                camlist.extend(scene_info.test_cameras)
            if scene_info.train_cameras:
                camlist.extend(scene_info.train_cameras)
            for id, cam in enumerate(camlist):
                json_cams.append(camera_to_JSON(id, cam))
            with open(os.path.join(self.model_path, "cameras.json"), 'w') as file:
                json.dump(json_cams, file)
```

**Comment by tongji-rkr** [comment]
> In functions `self.create_from_pcd` and `self.load_ply_sparse_gaussian`, the method of determining `self.levels` makes a difference. Here is the code.
> 
> In `self.load_ply_sparse_gaussian`,
> 
> ```
> self.levels = torch.max(self._level) - torch.min(self._level) + 1
> ```
> 
> while in `self.create_from_pcd`, the adaptive level selection mechanism determines `self.levels` as
> 
> ```
> self.levels = torch.round(torch.log2(dist_max/dist_min)/math.log2(self.fork)).int().item() + 1
> ```
> 
> This seems to assign different values to `self.levels` for the same model when initializing and reloading a `gaussian_model`, which is unexpected.
I admit that there might be some variation, but `self.levels` doesn't affect the rendering result, and it helps to avoid rendering a single level without content. Of course, saving it to `.ply` files should be a better solution.

**Comment by tongji-rkr** [comment]
> I also noticed something strange. `Scaffold-GS` and `Octree-GS` utilized a "warmup" training style. But in `Scaffold-GS`, setting `warmup` will replace the input point cloud with an already-trained point cloud, while in `Octree-GS` similar mechanism is missing in `scene/__init__.py: line 55-66`. However, setting `warmup` does promote the visual performance of `Octree-GS`.

The operations are different here, but the results should be the same

**Comment by ZhiyeTang** [comment]
> > I also noticed something strange. `Scaffold-GS` and `Octree-GS` utilized a "warmup" training style. But in `Scaffold-GS`, setting `warmup` will replace the input point cloud with an already-trained point cloud, while in `Octree-GS` similar mechanism is missing in `scene/__init__.py: line 55-66`. However, setting `warmup` does promote the visual performance of `Octree-GS`.
> 
> The operations are different here, but the results should be the same. `Scaffold-GS` writes the contents of the file to `input.ply`, `Octree-GS` reads the `.ply` file to `scene_info.point_cloud` in `scene/dataset_readers.py`, and then saves it to `input.ply`. The results should be the same.

I see, thanks a lot! 🙇‍
